lib/linter/node-event-generator.js

```diff
@@ -293,6 +308,12 @@ class NodeEventGenerator {
      * @returns {void}
      */
     applySelector(node, selector) {
+
+        if (this.knownTypes.has(selector.rawSelector)) {
+            this.emitter.emit(selector.rawSelector, node);
+            return;
+        }
```

Comment:
We could add a property `isIdentifier: parsedSelector.type === "identifier"` to [selectors](https://github.com/eslint/eslint/blob/880a4317b949e575a4a6c5e8baaba1eea7674cc6/lib/linter/node-event-generator.js#L216-L223) and just check `if (selector.isIdentifier)` here.

Reply:
Sorry, I’m missing some context here. Why would we do that?

Reply:
For performance reasons. Whether the selector is an identifier is already known at the parse time.

Reply:
Yes, at this point if it's an identifier then it certainly matches.

Reply:
Got it. Updated.